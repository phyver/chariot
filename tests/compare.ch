:set use_priorities false

:echo "TRUE means that the first pattern is an approximation of the second"

:echo "expect TRUE"
:test compare f x => g x and  f x => g x

:echo "expect FALSE"
:test compare f x => g (S x) and  f x => g x

:echo "expect FALSE"
:test compare f x => g x and  f x => g (S x)

:echo "expect FALSE"
:test compare f (S x) => g x and  f x => g x

:echo "expect FALSE"
:test compare f x => g x and  f (S x) => g x


:set depth 0

:echo "expect TRUE"
:test compare f (S x) => g (S x) and  f (S x) => g (S x)

:echo "expect TRUE"
:test compare f (S x) => g (S (S x)) and  f (S x) => g (S x)

:echo "expect FALSE"
:test compare f (S x) => g (S x) and  f (S x) => g (S (S x))

:set depth 1

:echo "expect TRUE"
:test compare f (S x) => g (S x) and  f (S x) => g (S x)

:echo "expect TRUE"
:test compare f (S x) => g (S (S x)) and  f (S x) => g (S x)

:echo "expect FALSE"
:test compare f (S x) => g (S x) and  f (S x) => g (S (S x))


:echo "expect TRUE"
:test compare f x y => g x y  and  f x y => g x y

:echo "expect TRUE"
:test compare f x y => g x y  and  f x => g x

:echo "expect TRUE"
:test compare f x => g x  and  f x y => g x y

:echo "expect FALSE"
:test compare f x y => g x y  and  f x y => g x

:echo "expect FALSE"
:test compare f x y => g x y  and  f x y => g x x

:echo "expect FALSE"
:test compare f x y => g x y  and  f x y => g y x

:echo "expect FALSE"
:test compare f x y => g x y  and  f x y => g x (S y)

:echo "expect FALSE"
:test compare f (S x) y => g x y  and  f x y => g x y


:set depth 0

:echo "expect FALSE"
:test compare f x y => g x y  and  f x y => g x (S y)

:echo "expect FALSE"
:test compare f (S x) y => g x y  and  f x y => g x y

:echo "expect TRUE"
:test compare f x y => g (S x) y  and  f x y => g x y

:echo "expect FALSE"
:test compare f x y => g x y  and  f x (S y) => g x y

:echo "expect TRUE"
:test compare f x y => g (S x) y  and  f (S x) y => g x y

:echo "expect FALSE"
:test compare  f (S x) y => g y x and f x y => g (S x) y


:echo "expect FALSE"
:test compare f (S x) => g x  and  f x => g x

:echo "expect TRUE"
:test compare f x => g (S x)  and  f x => g x


