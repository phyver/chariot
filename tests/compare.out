-- TRUE means that the first pattern is an approximation of the second
-- expect TRUE
--   f, x => g, x    ≥    f, x => g, x
--         TRUE

-- expect FALSE
--   f, x => g, S⁼ x    ≥    f, x => g, x
--         FALSE

-- expect FALSE
--   f, x => g, x    ≥    f, x => g, S⁼ x
--         FALSE

-- expect FALSE
--   f, S⁼ x => g, x    ≥    f, x => g, x
--         FALSE

-- expect FALSE
--   f, x => g, x    ≥    f, S⁼ x => g, x
--         FALSE

-- expect TRUE
--   f, x₁ => g, <0>⁼ x₁    ≥    f, x₁ => g, <0>⁼ x₁
--         TRUE

-- expect TRUE
--   f, x₁ => g, <1>⁼ x₁    ≥    f, x₁ => g, <0>⁼ x₁
--         TRUE

-- expect FALSE
--   f, x₁ => g, <0>⁼ x₁    ≥    f, x₁ => g, <1>⁼ x₁
--         FALSE

-- expect TRUE
--   f, S⁼ x => g, S⁼ x    ≥    f, S⁼ x => g, S⁼ x
--         TRUE

-- expect TRUE
--   f, S⁼ x => g, S⁼ <1>⁼ x    ≥    f, S⁼ x => g, S⁼ x
--         TRUE

-- expect FALSE
--   f, S⁼ x => g, S⁼ x    ≥    f, S⁼ x => g, S⁼ <1>⁼ x
--         FALSE

-- expect TRUE
--   f, x, y => g, x, y    ≥    f, x, y => g, x, y
--         TRUE

-- expect TRUE
--   f, x, y => g, x, y    ≥    f, x => g, x
--         TRUE

-- expect TRUE
--   f, x => g, x    ≥    f, x, y => g, x, y
--         TRUE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, y => g, x
--         FALSE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, y => g, x, x
--         FALSE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, y => g, y, x
--         FALSE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, y => g, x, S⁼ y
--         FALSE

-- expect FALSE
--   f, S⁼ x, y => g, x, y    ≥    f, x, y => g, x, y
--         FALSE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, y => g, x, <1>⁼ y
--         FALSE

-- expect FALSE
--   f, x₁, y => g, <-1>⁼ x₁, y    ≥    f, x, y => g, x, y
--         FALSE

-- expect TRUE
--   f, x, y => g, <1>⁼ x, y    ≥    f, x, y => g, x, y
--         TRUE

-- expect FALSE
--   f, x, y => g, x, y    ≥    f, x, x₁ => g, x, <-1>⁼ x₁
--         FALSE

-- expect TRUE
--   f, x, y => g, <1>⁼ x, y    ≥    f, x₁, y => g, <-1>⁼ x₁, y
--         TRUE

-- expect FALSE
--   f, x₁, y => g, y, <-1>⁼ x₁    ≥    f, x, y => g, <1>⁼ x, y
--         FALSE

-- expect FALSE
--   f, x₁ => g, <-1>⁼ x₁    ≥    f, x => g, x
--         FALSE

-- expect TRUE
--   f, x => g, <1>⁼ x    ≥    f, x => g, x
--         TRUE

